### DISCUSSION TITLE
What is the difference between buffer server messages?

### DISCUSSION THREAD

**Discussion by azizsayfiddin** [squidowl/halloy#315] [Q&A] [answered]
What is the difference between buffer.server_messages.join, buffer.server_messages.part, buffer.server_messages.quit, and buffer.server_messages.topic?

**Comment by casperstorm** [comment, marked as answer]
It is messages from the server which you see in a buffer. So, `join` is when a user joins a channel, `part` is when they leave and so on.

You can control these to eg. hide them if you they are irrelevant to you.

- join: user joins a channel
- part: user leaves a channel
- quit: user disconnects from server
- topic: channel topic
